.evergreen/serverless/create-instance.sh

```diff
@@ -1,21 +1,20 @@
 #!/bin/bash
 
 set -o errexit
-set +o xtrace # disable xtrace to ensure credentials aren't leaked
-
-# INSTANCE_NAME has the following limitations:
-# https://docs.atlas.mongodb.com/reference/atlas-limits/#label-limits
-INSTANCE_NAME="$RANDOM-DRIVERTEST"
-
-# Set the LOADBALANCED environment variable to "ON" to opt-in to
-# testing load balanced serverless instances.
-if [ "$LOADBALANCED" = "ON" ]; then
-    BACKING_PROVIDER_NAME="AWS"
-    INSTANCE_REGION_NAME="US_EAST_1"
-else
-    BACKING_PROVIDER_NAME="GCP"
-    INSTANCE_REGION_NAME="CENTRAL_US"
-fi
+set +o xtrace # Disable xtrace to ensure credentials aren't leaked
+
+# Supported environment variables:
+#
+#   SERVERLESS_INSTANCE_NAME    Optional. Serverless instance to create (defaults to a random name).
+#   SERVERLESS_DRIVERS_GROUP    Required. Atlas group for drivers testing.
+#   SERVERLESS_API_PUBLIC_KEY   Required. Public key for Atlas API request.
+#   SERVERLESS_API_PRIVATE_KEY  Required. Private key for Atlas API request.
+#
+# On success, this script will output serverless-expansion.yml with the
+# following expansions:
+#
+#   SERVERLESS_URI            SRV connection string for newly created instance
```

Comment:
I considered using `MONGODB_URI`, but this prefix is consistent `SERVERLESS_INSTANCE_NAME` and removes ambiguity. The previous version of this script set many environment variables (e.g. `MONGODB_URI`, `MONGODB_SRV_URI`, `SINGLE_ATLASPROXY_SERVERLESS_URI`), so changing to something new seemed like a clean break (even though we still define the old vars for BC).